examples/c/counter.c

```diff
@@ -114,284 +60,277 @@ void* imports_build() {
+  assert_svm_result(res, err, "svm_app_receipt_gas");
 
-    free((void*)host_ctx.bytes);
-    svm_byte_array_destroy(receipt);
-    svm_byte_array_destroy(err);
+  // Reclaim resources.
+  free(host_ctx.bytes);
+  free(creator.bytes);
+  free(ctor_args.values);
 
-    return (deploy_template_t) { .template_addr = template_addr };
+  return (spawn_app_result_t) {
+    .receipt = receipt,
+    .app_addr = app_addr,
+    .init_state = init_state,
+    .gas_used = gas_used
+  };
 }
 
-spawned_app_t simulate_spawn_app(void* runtime, svm_byte_array bytes, svm_byte_array creator) {
+exec_app_result_t simulate_exec_app(void* runtime, svm_byte_array app_addr, svm_byte_array app_state, uint32_t func_idx, svm_value_array func_args) {
   svm_result_t res;
-  svm_byte_array err; 
-
-  res = svm_validate_app(runtime, bytes, &err);
-  if (res == SVM_FAILURE) {
-    printf("`svm_validate_app` failure:\n");
-    print_err(err);
-    exit(1);
-  }
+  svm_byte_array err;
 
-  svm_byte_array host_ctx = host_ctx_empty_bytes();
-  svm_byte_array receipt; 
+  // Encode.
+  svm_byte_array app_tx;
+  uint32_t version = 0;
+  svm_byte_array func_buf = empty_byte_array();
+  res = svm_encode_app_tx(&app_tx, version, app_addr, func_idx, func_buf, func_args, &err);
+  assert_svm_result(res, err, "svm_encode_app_tx");
 
-  res = svm_spawn_app(&receipt, runtime, bytes, creator, host_ctx, 1, &err);
+  // Validate.
+  res = svm_validate_tx(&app_addr, runtime, app_tx, &err);
+  assert_svm_result(res, err, "svm_validate_tx");
 
-  if (res == SVM_FAILURE) {
-     printf("`svm_spawn_app` failure:\n");
-     print_err(err);
-     exit(1);
+  // Execute.
+  svm_byte_array host_ctx = host_ctx_empty_bytes();
+  svm_byte_array receipt;
+  res = svm_exec_app(&receipt, runtime, app_tx, app_state, host_ctx, GAS_METERING, GAS_LIMIT, &err);
+  assert_svm_result(res, err, "svm_exec_app");
+  assert(res == svm_exec_receipt_status(receipt, &err));
+
+  // Extract: new state.
+  svm_byte_array new_state;
+  res = svm_exec_receipt_state(&new_state, receipt, &err);
+  assert_svm_result(res, err, "svm_exec_receipt_state");
+    for (int i = 0; i < new_state.length; i++) {
+    printf("%02x", new_state.bytes[i]);
   }
 
-  svm_byte_array app_addr, init_state;
-  svm_app_receipt_state(&init_state, receipt);
-  svm_app_receipt_addr(&app_addr, receipt);
-
-  printf("Spawned App successfully\n");
-  print_addr(app_addr);
-  print_state(init_state);
-
-  svm_byte_array_destroy(host_ctx);
-
-  spawned_app_t spawned = {
-    .app_addr = app_addr,
-    .init_state = init_state
+  // Extract: returns.
+  svm_value_array returns;
+  res = svm_exec_receipt_returns(&returns, receipt, &err);
+  assert_svm_result(res, err, "svm_exec_receipt_returns");
+
+  // Extract: gas used.
+  uint64_t gas_used;
+  res = svm_exec_receipt_gas(&gas_used, receipt, &err);
+  assert_svm_result(res, err, "svm_app_receipt_gas");
+
+  // Reclaim resources.
+  free(host_ctx.bytes);
+
+  return (exec_app_result_t) {
```

Comment:
I'm not sure the surrounded braces are required here.

Reply:
I got compilation error without it:
```
unexpected type name 'exec_app_result_t': expected expression
```